program/src/instructions/shared_accounts.rs

```diff
@@ -114,6 +114,7 @@ pub struct TradeShared<'info> {
             pool.pool_id.as_ref(),
         ],
         bump = pool.bump[0],
+        has_one = owner @ ErrorCode::WrongOwner,
```

Comment:
Hmmm I removed this b/c it was redundant with the seeds.

otherwise to be consistent we'd have to add this to all the ixs again...

Reply:
I think a has_one constraint is better than a seeds here if we're only going to have one, but for consistency I removed it.